### ISSUE TITLE
Panic in network policy manager.

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#10460]
Rancher versions: Build from https://github.com/alena1108/cattle - "region" branch
network policy manager - leodotcloud/network-policy-manager:regions

Steps to reproduce the problem:Had following setup:
Region1 with environments - "test1","test2"
There were 2 hosts in both the environments.
"network policy manager"  was deployed with "allow within linked services" and deny for all other policies.

Deleted both hosts in both enviroenments and added 2 new host.

network policy manager is not able to start on the newly added hosts anymore.

Following panic is seen repeatedly.
```
12/1/2017 4:04:11 PMtime="2017-12-02T00:04:11Z" level=info msg="Waiting for metadata"
12/1/2017 4:04:11 PMpanic: runtime error: index out of range
12/1/2017 4:04:11 PM
12/1/2017 4:04:11 PMgoroutine 14 [running]:
12/1/2017 4:04:11 PMpanic(0x7c15a0, 0xc42000c0a0)
12/1/2017 4:04:11 PM	/usr/local/go/src/runtime/panic.go:500 +0x1a1
12/1/2017 4:04:11 PMgithub.com/rancher/network-policy-manager/policy.(*watcher).fetchNetworkInfo(0xc4200768c0, 0x15, 0xc4200c8ac0)
12/1/2017 4:04:11 PM	/go/src/github.com/rancher/network-policy-manager/policy/watcher.go:227 +0x612
12/1/2017 4:04:11 PMgithub.com/rancher/network-policy-manager/policy.(*watcher).fetchInfoFromMetadata(0xc4200768c0, 0x14, 0xc420043d58)
12/1/2017 4:04:11 PM	/go/src/github.com/rancher/network-policy-manager/policy/watcher.go:1181 +0x105e
12/1/2017 4:04:11 PMgithub.com/rancher/network-policy-manager/policy.(*watcher).onChange(0xc4200768c0, 0xc420051b00, 0x24, 0x1, 0x1)
12/1/2017 4:04:11 PM	/go/src/github.com/rancher/network-policy-manager/policy/watcher.go:1193 +0xf8
12/1/2017 4:04:11 PMgithub.com/rancher/network-policy-manager/policy.(*watcher).onChangeNoError(0xc4200768c0, 0xc420051b00, 0x24)
12/1/2017 4:04:11 PM	/go/src/github.com/rancher/network-policy-manager/policy/watcher.go:185 +0x11b
12/1/2017 4:04:11 PMgithub.com/rancher/network-policy-manager/policy.(*watcher).(github.com/rancher/network-policy-manager/policy.onChangeNoError)-fm(0xc420051b00, 0x24)
12/1/2017 4:04:11 PM	/go/src/github.com/rancher/network-policy-manager/policy/watcher.go:160 +0x3e
12/1/2017 4:04:11 PMgithub.com/rancher/network-policy-manager/vendor/github.com/rancher/go-rancher-metadata/metadata.(*client).OnChange.func1(0xc420051b00, 0x24)
12/1/2017 4:04:11 PM	/go/src/github.com/rancher/network-policy-manager/vendor/github.com/rancher/go-rancher-metadata/metadata/change.go:36 +0x5f
12/1/2017 4:04:11 PMgithub.com/rancher/network-policy-manager/vendor/github.com/rancher/go-rancher-metadata/metadata.(*client).onChangeFromVersionWithError(0xc4200512c0, 0x8183e2, 0x4, 0x5, 0xc4200c9f50, 0x0, 0x0)
12/1/2017 4:04:11 PM	/go/src/github.com/rancher/network-policy-manager/vendor/github.com/rancher/go-rancher-metadata/metadata/change.go:25 +0x1e5
12/1/2017 4:04:11 PMgithub.com/rancher/network-policy-manager/vendor/github.com/rancher/go-rancher-metadata/metadata.(*client).OnChange(0xc4200512c0, 0x5, 0xc42000d750)
12/1/2017 4:04:11 PM	/go/src/github.com/rancher/network-policy-manager/vendor/github.com/rancher/go-rancher-metadata/metadata/change.go:40 +0xa6
12/1/2017 4:04:11 PMcreated by github.com/rancher/network-policy-manager/policy.Watch
12/1/2017 4:04:11 PM	/go/src/github.com/rancher/network-policy-manager/policy/watcher.go:160 +0x300
```

**Comment by leodotcloud** [comment]
Fixed in `rancher/network-policy-manager:v0.2.5`

**Comment by sangeethah** [comment]
Tested with rancher/network-policy-manager:v0.2.5 .

Steps to reproduce the problem:Had following setup:
Region1 with environments - "test1","test2"
There were 2 hosts in both the environments.
"network policy manager" was deployed with "allow within linked services" and deny for all other policies.

Deleted both hosts in both enviroenments and added 2 new host.

network policy manager is able to start successfully on the newly added hosts.
Cross host communication works as expected.
